frontend/src/BlogDetail/BlogDetail.js

```diff
@@ -1,80 +1,79 @@
-import React from 'react';
-import NavBar from '../Navbar/NavBar';
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import NavBar from "../Template/NavBar"
+import Footer from "../Template/Footer"
+import API from "../API";
+import Comments from "./Comments"
 
-function BlogDetail() {
-	return (
-		<div>
-			<div>
-				<NavBar />
-			</div>
-			<div className='ui raised very padded text container'>
-				<div>
-					<h4 className='ui left aligned sub header'>Date Created</h4>
-					<h1 className='ui left aligned header'>Article Title</h1>
-				</div>
-				<div className='ui hidden divider'></div>
-				<div className='ui left aligned text container segment '>
-					<div className='ui placeholder'>
-						<div className='image header'>
-							<div className='line'></div>
-							<div className='line'></div>
-						</div>
-						<div className='paragraph'>
-							<div className='line'></div>
-							<div className='line'></div>
-							<div className='line'></div>
-							<div className='line'></div>
-							<div className='line'></div>
-						</div>
-					</div>
-				</div>
+const BlogDetail = () => {
+  const [post, setPost] = useState([]);
+  const { slug: postSlug } = useParams();
 
-				<div className='form container'>
-					<div className='ui form'>
-						<div className='field'>
-							<textarea> What are your thoughts?</textarea>
-						</div>
-					</div>
-				</div>
-				<div className='ui hidden divider'></div>
-				<div>
-					<button className='ui right floated blue button'>Comment </button>
-				</div>
-				<div className='ui hidden section divider'></div>
-				<div className='ui comments'>
-					<h3 className='ui dividing header'>Comments</h3>
-					<div className='comment'>
-						<div className='avatar'>
-							{/* <img src="/images/avatar/small/matt.jpg"> */}
-						</div>
-						<div className='content'>
-							<div className='author'>Matt</div>
-							<div className='metadata'>
-								<span className='date'>Today at 5:42PM</span>
-							</div>
-							<div className='text'>How artistic!</div>
-						</div>
-					</div>
-					<div className='comment'>
-						<div className='avatar'>
-							{/* <img src="/images/avatar/small/elliot.jpg"> */}
-						</div>
-						<div className='content'>
-							<div className='author'>Elliot Fu</div>
-							<div className='metadata'>
-								<span className='date'>Yesterday at 12:30AM</span>
-							</div>
-							<div className='text'>
-								<p>
-									This has been very useful for my research. Thanks as well!
-								</p>
-							</div>
-						</div>
-					</div>
-				</div>
-			</div>
-		</div>
-	);
-}
+
+  useEffect(() => {
+    API.get(`/posts/${postSlug}/`)
+      .then((res) => setPost(res.data))
+      .catch((err) => {
+        console.log(err.response.status);
+        if (err.response.status === 404) {
+          window.location.href = "/notfound";
+          return;
+        }
```

Comment:
- [ ] remove console.log
- [ ] use react's useHistory hook instead
https://www.telerik.com/blogs/programmatically-navigate-with-react-router#:~:text=push()%20Method,Component%20prop%20to%20a%20Route.

Reply:
I tried using useHistory sir but I think that useHistory is not available in react-router-dom v6, it suggests in this link sir that I will useNavigate() instead. Should I useNavigate()?

https://stackoverflow.com/questions/62861269/attempted-import-error-usehistory-is-not-exported-from-react-router-dom

Reply:
navigate is fine